ambry-api/src/main/java/com/github/ambry/config/CompressionConfig.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.config;
+
+import com.github.ambry.utils.Utils;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Configuration parameters required by a CompressionService.
+ * Configurations are read-only.  The only way to alter the values is by modifying the config files.
+ * The configuration applies to compression only.  Decompression is automatic and has no configuration.
+ */
+public class CompressionConfig {
+
+  // Boolean whether compression is enabled in PUT operation.
+  private static final String CONFIG_COMPRESSION_ENABLED = "router.compression.enabled";
+  private static final boolean DEFAULT_COMPRESSION_ENABLED = true;
+
+  // Whether to skip compression if content-encoding present.
+  private static final String CONFIG_SKIP_CONTENT_ENCODING = "router.compression.skip.content.encoding";
+  private static final boolean DEFAULT_SKIP_CONTENT_ENCODING = true;
+
+  // A comma-separated list of compressible fully content-types, like "text/javascript, text/log; Charset=UTF8"
+  private static final String CONFIG_CONTENT_TYPES_COMPRESSIBLE = "router.compression.content.types.compressible";
+  private static final String DEFAULT_CONTENT_TYPES_COMPRESSIBLE = null;
+
+  // A comma-separated list of incompressible fully content-types, like "text/javascript, text/log; Charset=UTF8"
+  private static final String CONFIG_CONTENT_TYPES_INCOMPRESSIBLE = "router.compression.content.types.incompressible";
```

Comment:
what happens if text/javascript is present in both configs - compressible and incompressible ?

Reply:
If same setting is in both incompressible and compressible, incompressible takes priority because the map from content-type to boolean adds compressible content-type first, followed by incompressible content-types.